drivers/libusb.c

```diff
@@ -47,9 +47,6 @@ upsdrv_info_t comm_upsdrv_info = {
 
 #define MAX_REPORT_SIZE         0x1800
 
-/* Used for Powervar UPS to make sure we use the right interface in the Composite device */
-static int hid_rep_index = 0;
-
 static void libusb_close(usb_dev_handle *udev);
 
 /*! Add USB-related driver variables with addvar().
```

Comment:
Follow-up idea/question: would it make sense to add a driver configuration tunable here, for people to try a non-zero interface number? I suppose over time there can be more HID devices exposing an UPS PDC in a way that current NUT of the day does not know how to find...

Also, is it possible to detect and report if a device has more than one (the zero-numbered) interface, especially if no UPS support is found on #0, as a hint for log readers to try looking further?

Reply:
e.g. looking at the user report log from https://alioth-lists.debian.net/pipermail/nut-upsuser/2021-May/012410.html / https://alioth-lists.debian.net/pipermail/nut-upsuser/attachments/20210518/30968f29/attachment.htm with errors like:

````
           0.056845     Trying to match device
           0.057251     Device matches
           0.057334     failed to claim USB device: could not claim interface 0: No such file or directory
           0.057475     failed to detach kernel driver from USB device: could not detach kernel driver from interface 0: No route to host
           0.057554     failed to claim USB device: could not claim interface 0: No such file or directory
           0.057600     failed to detach kernel driver from USB device: could not detach kernel driver from interface 0: No route to host
           0.057768     failed to claim USB device: could not claim interface 0: No such file or directory
           0.057825     failed to detach kernel driver from USB device: could not detach kernel driver from interface 0: No route to host
           0.057870     failed to claim USB device: could not claim interface 0: No such file or directory
           0.058006     failed to detach kernel driver from USB device: could not detach kernel driver from interface 0: No route to host
           0.058050     Can't claim USB device [0764:0601]: could not detach kernel driver from interface 0: No route to host
````

I can't prove or disprove quickly if there is some new device revision with "interface 0" used for something while another interface got introduced for UPS activity. Or if that is just some system setup error (permissions, libusb version, ...) on that user's side...

Reply:
I thought about this. Placing interface and end points  in ups.conf would be better of course. Another alternative - Win and OSX hid ups drivers find the right ones without any settings. In theory, one can cycle through the interfaces and determine the one with UPS by reading the descriptor.

Reply:
That "lockpicking" would be a nice option too (though something with a toggle to enable/disable, just in case). I do not know if there is a way to "know" as a libusb consumer that an interface actually exists on a device, and if the range is contiguous (non-replying interface means we stop iterating). And if the errors like in the log above are related to this and can be used as hints for the iterator :)

Reply:
@jimklimov agreed, I suggest we can do in the next PR in order not to pack too many changes in this one.